File: sqlite-android-plugin/src/main/java/com.alecstrong.sqlite.android/lang/SqliteLanguage.java
package com.alecstrong.sqlite.android.lang;

import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import org.jetbrains.annotations.NotNull;

public class SqliteLanguage extends Language {
	public static SqliteLanguage INSTANCE = new SqliteLanguage();

	public SqliteLanguage() {
		super("Sqlite", "text/sqlite");
		SyntaxHighlighterFactory.LANGUAGE_FACTORY.addExplicitExtension(this, new SingleLazyInstanceSyntaxHighlighterFactory() {
			@NotNull
			protected SyntaxHighlighter createHighlighter() {
				return new SqliteHighlighter();
			}
		});
	}
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/squareup/sqlite/android/lang/SqliteLanguage.kt
package com.squareup.sqlite.android.lang

import com.intellij.lang.Language
import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory

class SqliteLanguage : Language("Sqlite", "text/sqlite") {
  init {
    SyntaxHighlighterFactory.LANGUAGE_FACTORY.addExplicitExtension(this,
        object : SingleLazyInstanceSyntaxHighlighterFactory() {
          override fun createHighlighter() = SqliteHighlighter()
        })
  }

  companion object {
    var INSTANCE = SqliteLanguage()
  }
}
